### DISCUSSION TITLE
Problem BSD33 16A Power Monitoring Plug + Tasmota

### DISCUSSION THREAD

**Discussion by GeorgeWatch** [arendst/Tasmota#16109] [Q&A] [answered]
Hi,
I have a problem with BSD33 16A Power Monitoring Plug PCB BSD34-202101 + flash Tasmota. 
I replaced the original W82S board with an ESP-02S board. I loaded [Tasmota](https://templates.blakadder.com/BSD33-16A.html) and typed in the console:
  template {"NAME":"Generic","GPIO":[0,255,0,131,134,132,0,0,21,17,56,0,0],"FLAG":0,"BASE":18}.
I found that the button on the body of the BSD33 16A does not work.
The relay can be turned on/off on the web interface, but vice versa:
Web console toogle click: On=relay BSD33 is Off, console shows Voltage=294V.
Web console toogle click: Off=relay BSD33 is On, console shows Voltage=0V.
After connecting to the mains, the relay clicks twice.

Can something be done about it?
Thank you.
G.W.

**Comment by barbudor** [comment, marked as answer]
It happens that whole keeping the same model number, the hardware design changes
So you will have to tweak and find the right GPIO configuration
The easiest is the relay which seems to be inverted, so change Gpio12 from Relay to Relay_i (in menu Configuration, Template)

For the button not working, you need to move the 'Button 1' définition from GPIO13 to any other free GPIO until it works

**Reply by GeorgeWatch** [reply]
Thank you for answer. I'm trying it for the first time. When I swapped the Relay behind the Relay_i, it behaves the same way the other way around. Only the voltage stopped showing. With Relay_i On/Off, the voltage there is still 0V ... 
Uff ...

**Comment by BrAiNeeBug** [comment]
I have a Cozylife Tasmota Mode1 BSD33 wich was shipped wis wrong template (doing random restarts etc), also all BSD33 Pluds i found, the Profiles didn work.
I found for me the perfect working template wich was from a SP-20 Device.
{"NAME":"BSD33-COZY","GPIO":[0,0,0,32,2720,2656,0,0,2624,320,224,0,0,0],"FLAG":0,"BASE":18}

Regards BrAiNee
